mmv1/templates/terraform/examples/firestore_database.tf.erb

```diff
@@ -27,6 +27,7 @@ resource "google_firestore_database" "<%= ctx[:primary_resource_id] %>" {
   concurrency_mode                  = "OPTIMISTIC"
   app_engine_integration_mode       = "DISABLED"
   point_in_time_recovery_enablement = "POINT_IN_TIME_RECOVERY_ENABLED"
+  delete_protection_state           = "DELETE_PROTECTION_ENABLED"
```

Comment:
Will this cause the database to not be deleted when the test completes?

Reply:
It will prevent the database from being deletable!

Meaning with `delete_protection_state` set to `DELETE_PROTECTION_ENABLED` this request would fail:

```
DELETE https://firestore.googleapis.com/v1/projects/not-a-project/databases/(default)
```

While this request would succeed when the `delete_protection_state` is set to `DELETE_PROTECTION_DISABLED`.
-- 
https://cloud.google.com/firestore/docs/manage-databases#delete_protection

It behaves similarly to the Cloud SQL instance deletion protection: https://cloud.google.com/sql/docs/mysql/deletion-protection#rest-v1
Or the Google Cloud Project liens deletion protection: https://cloud.google.com/resource-manager/docs/project-liens
-- 
Setting, unsettling, or changing this field has no direct impact on the deletion of the database.

So as I understand it, when you create a database with `delete_projection_state` set to `DELETE_PROTECTION_ENABLED`, you can't remove that resource in a single change request/PR.
You first have to set it to `DELETE_PROTECTION_DISABLED` and then you can remove the resource.

This is useful to prevent accidental deletions of Firestore databases.

Reply:
In that case I think we'll want to use `skip_test` to prevent this test from being run and leaving behind databases in our test environment. We should rely on a handwritten test that removes deletion protection before completing so that the test databases can be cleaned up.

Reply:
@trodge that makes sense!

I'm wondering. I've already added a custom test that checks if updating works for that field!
https://github.com/GoogleCloudPlatform/magic-modules/pull/8906/files#diff-0f0af92ffe258bbd03d14d44b699ac64106de5d46fed4db170cb2c37ab5a8546R92-R120

And, as I understand it, that test 
- creates a database with that field set
- verifies if the database was created properly with the delete protection enabled
- updates the value to disable the delete protection enabled
- verifies if the delete protection is disabled

So that test should cover those cases already, no?

Is there anything else I should add to the tests?
Besides adding the `skip_test` flag to the `delete_protection_state` field?

Reply:
@trodge I've added the `skip_test` flag to the field.

Reply:
`skip_test` is an attribute of the example, not the field. We'll want to skip the example test because the update test already covers everything it covers and we don't want to leave orphaned resources.

Reply:
@trodge sorry, misunderstood you there!

I did not want to skip the tests for `firestore_database` as a lot of other fields are being tested there.
So I've removed the `delete_protection_state` field from this standard test and added a custom example (which is being skipped) for the delete protection.

I took the opportunity to describe how to remove a database with delete protection enabled!